### ISSUE TITLE
IP Collision

### ISSUE THREAD

**Issue by joelworsham** [Varying-Vagrant-Vagrants/VVV#1605]
## Expected Behavior
vagrant is able to boot up

## Current Behavior
vagrant cannot boot up and provides an error. Here is full console:

```
__ __ __ __
\ V\ V\ V / Varying Vagrant Vagrants
 \_/\_/\_/  v2.2.2-develop-git::develop

Platform:   darwin13.4.0 shell:/bin/zsh
Vagrant:    2.1.1
VirtualBox: 5.2.8r121009

Docs:       https://varyingvagrantvagrants.org/
Contribute: https://github.com/varying-vagrant-vagrants/vvv
Dashboard:  http://vvv.test

==> default: Running triggers before reload ...
==> default: Running trigger: VVV Pre-Reload...
    default: Running: inline script
    default: Performing Database Backups
    ... lots of db backup (excluded) ...
==> default: [vagrant-hostsupdater] Removing hosts
==> default: Attempting graceful shutdown of VM...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: A newer version of the box 'ubuntu/trusty64' for provider 'virtualbox' is
==> default: available! You currently have version '20170110.0.0'. The latest is version
==> default: '20180722.0.0'. Run `vagrant box update` to update.
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
The specified host network collides with a non-hostonly network!
This will cause your specified IP to be inaccessible. Please change
the IP or name of your host only network so that it no longer matches that of
a bridged or non-hostonly network.

Bridged Network Address: '192.168.50.0'
Host-only Network 'en6: Apple USB Ethernet Adapter': '192.168.50.0'
```

## Possible Solution
Change IP in vagrantfile. Unfortunately, doing so seemed to "mess everything up". Once the machine booted (which it was able to), nothing was accessible in the browser. Also, this creates a conflict where I can't pull latest from git.

## Your Environment

* VVV version: latest `develop`. Pulled just before posting this. v2.2.2-develop-git::develop
* VVV Git Branch: develop
* Vagrant version: 2.1.1
* VM Provider name: VirtualBox
* VM Provider version: 5.2.8r121009
* Operating System and version: Mac OSX 10.13.6 (17G65)

**Comment by tomjn** [comment]
You can change this via `vvv-custom.yml`:

```yaml
vm_config:
    private_network_ip: 1.2.3.4
```

You will need to reprovision afterwards.

note that this might be caused by having multiple VVV instances running. Let me know if this helps

**Comment by joelworsham** [comment]
Thanks for the info!

I changed the Vagrantfile back to default and added that code to the `vvv-custom.yml` file. I made the IP `192.168.51.0`. I then ran `vagrant up --provision`.

It was able to provision, but now every single local site just shows "It works!"

![](https://d.pr/i/k3z6vE+)

**Comment by tomjn** [comment]
it doesn't sound like provisioning was succesful, `It Works!` isn't what Nginx outputs by default in VVV, you should be seeing the dashboard.

My recommendation is to double check that you do not have another local environment running on port 80 that could be conflicting. In the past when this has happened the user has Homestead or another vagrant environment running which used Apache, and it would randomly show either VVV or the other

Otherwise, I would reprovision from scratch and post the log file here.

Additionally, what do you see when you visit `http://192.168.51.0:80` And why did you choose such a similar IP?

**Comment by joelworsham** [comment]
I've never used another local environment on my machine (to my knowledge), so I'm guessing it wouldn't be that, though I can try to check deeper.

When I try to visit the IP in my browser, it does not load:

![](https://d.pr/i/Fg0fAx+)

I arbitrarily set the IP to something similar. I do not have much of any knowledge about local IP protocols, so I figured I'd make it different, but close. I can try a different one if you think it could help. I assumed `1.2.3.4` was just an example?

**Comment by joelworsham** [comment]
Not sure if this helps, but I ran a port scan on that IP and got this:

![](https://d.pr/i/vlmVz3+)

**Comment by tomjn** [comment]
Did you ever identify what was using the IP VVV originally tried to use? I would destroy the VM and start again with a fresh provision from scratch, and make sure to grab a full copy of the provision log

**Comment by tomjn** [comment]
And anything under `192.168.x.x` that's unoccupied will do

**Comment by joelworsham** [comment]
Well, the only change between working for 2 years and now is I got a new router. I suspect this because if you look at the original output I shared, it blames my ethernet adapter (I have a USB - Ethernet for a wired direct to the modem)

```
Bridged Network Address: '192.168.50.0'
Host-only Network 'en6: Apple USB Ethernet Adapter': '192.168.50.0'
```

Sorry for my naivety, but can I destroy my VM and not lose all of my data?

**Comment by tomjn** [comment]
@joelworsham you mean this isn't a fresh VM on its first provisions? Have you tried a different IP?

> Well, the only change between working for 2 years and now is I got a new router.

Your router may be using a different IP range than your previous one.

> Sorry for my naivety, but can I destroy my VM and not lose all of my data?

You don't, but that your VM was previously working is brand new information. Turn VVV off, clear out your hosts file, and verify that the IP you chose is indeed not in use. Then, find a new IP. **Then reprovision.** If `192.168.50.4` is in use by something, then moving to an IP that's very close is unwise.

E.g. if I have a local network and the 4th IP is occupied, moving to the 0th isn't a good idea as it's probably going to be assigned. Maybe the 200th or the 99999th IP would be a better choice, or an IP that isn't in the range the router assigns

In the meantime, VVV is meant to keep backups of the database on halt and up, check that those are intact, and you can use them to restore your databases.

Or, modify your router to assign IP ranges that aren't in the same IP range VVV uses

**Comment by tomjn** [comment]
Further research indicates that "It works!" is not an Nginx thing but an Apache thing.

Whatever is generating the "It Works!" it is not VVV

**Comment by tomjn** [comment]
The default nginx webpage contains a heading saying Welcome to Nginx, whatever you were loading, it was not VVV.

And by local dev environment, I don't just mean vagrant environments, I mean anything that serves on port 80, including ruby/go/python, docker containers, etc

**Comment by tomjn** [comment]
You need to identify what is binding to port 80 and acting as a web server and disable it, after which VVV will likely work fine as is

**Comment by LoreleiAurora** [comment]
@joeworsham please try changing the IP to 10.8.7.6 then reprovision.

What is the local IP of the host machine?

**Comment by joelworsham** [comment]
Hold on, I just ran `vagrant reload` with no changes it is now working...

I have no idea why, but it is working on that `10.8.7.6` IP. Thanks again for all the help!

**Comment by tomjn** [comment]
I'm going to close this for now, if it happens again feel free to reopen

**Comment by joelworsham** [comment]
Hey there,

So, since changing the IP, xdebug doesn't want to work. I am able to get it turned on, as it shows all of the settings and whatnot in vvv.dev/phpinfo, but projects that used to work with breakpoints just fine are no longer working.

I attempted to change`xdebug.remote_host` within `xdebug.ini` to the new IP and re-provision, but it still does not want to work...

Any ideas?

**Comment by tomjn** [comment]
@joelworsham you need to change it at both ends. PHPStorm will not connect to a different IP unless you tell it to, so you need to amend all the instructions as you read them with the new IP

Also, stop using `vvv.dev`, it will cause issues in the future if you aren't already running into them. Use `vvv.test` and `.test` domains

**Comment by tomjn** [comment]
I would also note that I'm unsure why you would need to change `xdebug.ini`, think of it this way:

 > Alice and Bob go for a stroll, but before their stroll begins, a notice appears telling Alice that her name is already taken. Alice is now known as Eve.
> 
> Bob makes a note of this by incorrectly changing his rules so that Bob only responds to messages intended for Eve, rather than messages intended for Bob
> 
> Eve tries to talk to Bob on their stroll, but Bob sees the conversation is intended for him, not Eve, and promptly ignores it

Or perhaps I've misunderstood, providing examples of the changes made would be helpful

**Comment by joelworsham** [comment]
Yes, I have been using vvv.test, I mis-typed. Whoops. I'll give this a shot! Thank you for the reply.

**Comment by joelworsham** [comment]
In PHP Storm I actually never enter the IP, I just enter mysite.test with either port 80 or 443 (it typically auto detects).

I tried entering the manual IP that comes through phpinfo(), and it still will not connect anymore.

I am calling `phpinfo()` right in my custom server to make sure it all lines up. Here is it showing the server address and port: 

![](https://d.pr/i/huHRet+)

here is some xdebug printed info:

![](https://d.pr/i/JcCW6K+)

I've tried the following configurations:

10.8.7.6:9000
10.8.7.6:80
10.8.7.6:443

(affinitynurse.test is the domain)
affinitynurse.test:9000
affinitynurse.test:80
affinitynurse.test:443

None work. This worked prior to changing the IP address that VVV uses. I really am at a loss :/
